lib/components/data/plots/CWMSPlot.jsx

```diff
@@ -12,77 +12,128 @@ const config_v2 = new Configuration({
 });
 const ts_api = new TimeSeriesApi(config_v2);
 
+
+const config_level = new Configuration({
+  headers: {
+    accept: "*/*",
+  },
+});
+
+const level_api = new LevelsApi(config_level)
+
+
+
 export default function CWMSPlot({
   tsids,
+  levels,
   office,
   begin,
   end,
   title,
-  fontSize,
-  unit = "EN",
+  unit,
+  timeseriesParams,
+  locationLevelParams,
+  layoutParams,
   className = "",
-  plotHeight = 550,
-  shapes = [],
-  autoSize = true,
-  annotations = [],
   responsive = true,
   loadingComponent = null,
-  layoutGrid = null,
 }) {
+
   const plotElement = useRef(null);
-  const [metaData, setMetaData] = useState(null);
-  const [plotTitle, setPlotTitle] = useState(null);
   const [plotTSIDs, setPlotTSIDs] = useState(null);
-  const [plotFontSize, setPlotFontSize] = useState(fontSize);
+  const [plotLevelIDs, setPlotLevels] = useState(null);
+
+
+  const layout = layoutParams
+
 
   useEffect(() => {
-    if (!title) {
-      setPlotTitle((Array.isArray(tsids) ? tsids[0] : tsids).split(".")[0]);
-    }
     if (!tsids.length)
       throw Error("You must specify one or more Timeseries IDs to plot.");
     if (!office) throw Error("You must specify a 3 letter ID for the office");
     if (typeof tsids == "string") {
       tsids = [tsids];
     }
     setPlotTSIDs(tsids);
-  }, [title, tsids, office]);
+    setPlotLevels(levels)
+  }, [title, tsids, levels, office]);
+
 
   const fetchData = async () => {
-    let promises = plotTSIDs.map(async (name) => {
+
+    let ts_promises = plotTSIDs.map(async (tsid) => {
       try {
+        // Currently, large page size calls are blocked, so the default of 500 is used
+        let pageSize = 500
+        // const delta = dayjs(end.value).diff(dayjs(start.value), "day", true)
+        // let interval = tsid.split(".")[3]
+        // if (interval.includes("Minute")) {
+        //   if (interval.includes("15")) { pageSize = delta * 100 }
+        //   if (interval.includes("1Minute")) { pageSize = delta * 1500 }
+        // }
+        // if (interval.includes("Hour")) { pageSize = delta * 10 }
+        // if (interval.includes("Day")) { pageSize = delta * 1 }
         return await ts_api.getCwmsDataTimeseries({
-          name,
-          end,
-          unit,
-          begin,
-          office,
-        });
+          name: tsid,
+          begin: begin,
+          end: end,
+          pageSize: pageSize,
+          office: office,
+          unit: unit
+        })
       } catch (error) {
-        if (error.response?.status === 404) {
-          console.warn(`Data for ${name} not found: 404`);
-          return null;
-        } else {
-          throw error;
-        }
+        console.error("Error fetching data:", error)
       }
-    });
-    let values = await Promise.all(promises);
-    let _data = { ts: {}, dates: [] };
-    // TODO: This should probably be the parameter not the units
+    })
+
+    let lev_promises = plotLevelIDs.map(async (item) => {
```

Comment:
This causes an error if a levels array isn't provided (which shouldn't be required).